### PR TITLE
Docs for ExternalJWTSigner

### PR DESCRIPTION
**Pull request by HarshalNeelkamal**
### Description

This is a placeholder PR

### Issue

- PR on `kubernetes/kubernetes`: https://github.com/kubernetes/kubernetes/pull/128190
- `kubernetes/enhancement` issue: https://github.com/kubernetes/enhancements/issues/740

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
Please sign CLA

**Comment by utkarsh-singh1** [comment]
@HarshalNeelkamal, make sure the changes you have pushed are yours, this PRs contains lots of commits from different authors.

**Comment by tengqm** [comment]
This PR should target the `dev-1.32` branch rather than `main`.

**Comment by AnshumanTripathi** [comment]
Hello @HarshalNeelkamal :wave:, 1.32 Docs Shadow here,
A quick reminder to get your PR ready for review before the deadline, Tuesday November 19th at 18:00 PDT. For additional information, refer to: [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review).
Thank you!

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 1]
```diff
@@ -544,6 +550,21 @@ Then, delete the Secret you now know the name of:
 kubectl -n examplens delete secret/example-automated-thing-token-zyxwv
 ```
 
+## External ServiceAccount token signing and key management
+
+{{< feature-state feature_gate_name="ExternalServiceAccountTokenSigner" >}}
+
+The kube-apiserver can be configured to use external-jwt-signer for token signing and key management.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:557]:
```suggestion
The kube-apiserver can be configured to use external signer for token signing and token verifying key management.
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 2]
```diff
@@ -544,6 +550,21 @@ Then, delete the Secret you now know the name of:
...
+This feature enables kubernetes distributions to integrate with key management solutions of their choice (eg: HSMs, cloud KMSes).
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:558]:
```suggestion
This feature enables kubernetes distributions to integrate with key management solutions of their choice (eg: HSMs, cloud KMSes) for service account credential signing and verification.
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 3]
```diff
@@ -544,6 +550,21 @@ Then, delete the Secret you now know the name of:
...
+To configure kub-apiserver to use external-jwt-signer set the `--service-account-signing-endpoint` flag to the location of a UDS on a filesystem, or
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:559]:
```suggestion
To configure kube-apiserver to use external-jwt-signer set the `--service-account-signing-endpoint` flag to the location of a Unix domain socket (UDS) on a filesystem, or
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 4]
```diff
@@ -544,6 +550,21 @@ Then, delete the Secret you now know the name of:
...
+be prefixed with an @ symbol and name a UDS in the abstract socket namespace.
+At the configured UDS, shall be an RPC server which implements [ExternalJWTSigner](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/externaljwt/apis/v1alpha1/api.proto).
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:561]:
```suggestion
At the configured UDS, shall be an RPC server which implements [ExternalJWTSigner](https://github.com/kubernetes/kubernetes/blob/release-1.32/staging/src/k8s.io/externaljwt/apis/v1alpha1/api.proto).
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 5]
```diff
@@ -544,6 +550,21 @@ Then, delete the Secret you now know the name of:
...
+The external-jwt-signer must be healthy and be ready to serve supported service account keys for the kube-apiserver to start.
+
+Check out [KEP-740](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/740-service-account-external-signing) for more details on ExternalJWTSigner.
+
+NOTE: The kube-apiserver flags --service-account-key-file and --service-account-signing-key-file will continue to be used for reading from files (unless, --service-account-signing-endpoint is set; They are mutually exclusive ways of supporting JWT signing and authentication)
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:566]:
```suggestion
NOTE: The kube-apiserver flags --service-account-key-file and --service-account-signing-key-file will continue to be used for reading from files unless --service-account-signing-endpoint is set; they are mutually exclusive ways of supporting JWT signing and authentication.
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates/external-service-account-token-signer.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+_build:
+  list: never
+  render: false
+
+stages:
+  - stage: alpha
+    defaultValue: false
+    fromVersion: "1.32"
+---
+Make the kube-apiserver use [ExternalJWTSigner](/docs/reference/access-authn-authz/service-account-admin#external-token-signer) for external token signing and key management.
```
Comment (liggitt) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/external-service-account-token-signer.md:13]: The feature gate doesn't *make* this happen, it *allows* configuring this.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/external-service-account-token-signer.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+Requires `ExternalServiceAccountTokenSigner` to be enabled.
```
Comment (liggitt) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/external-service-account-token-signer.md:14]: Drop this line, this file is documenting the feature gate itself

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 8]
```diff
@@ -319,6 +319,12 @@ Similarly, you must pass the corresponding public key to the `kube-apiserver`
 using the `--service-account-key-file` flag. The public key will be used to
 verify the tokens during authentication.
 
+{{< feature-state feature_gate_name="ExternalServiceAccountTokenSigner" >}}
+
+An alternate setup to setting `--service-account-private-key-file` and `--service-account-key-file` flags is
+to configure a external-jwt-signer for [External ServiceAccount token signing and key management](#external-erviceAccount-token-signing-and-key-management).
```
Comment (tengqm) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:325]:
```suggestion
to configure an external JWT signer for [external ServiceAccount token signing and key management](#external-erviceAccount-token-signing-and-key-management). 
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 9]
```diff
@@ -319,6 +319,12 @@ Similarly, you must pass the corresponding public key to the `kube-apiserver`
 using the `--service-account-key-file` flag. The public key will be used to
 verify the tokens during authentication.
 
+{{< feature-state feature_gate_name="ExternalServiceAccountTokenSigner" >}}
+
+An alternate setup to setting `--service-account-private-key-file` and `--service-account-key-file` flags is
+to configure a external-jwt-signer for [External ServiceAccount token signing and key management](#external-erviceAccount-token-signing-and-key-management). 
+Note that both are mutually exclusive setups and cannot be configured together.
```
Comment (tengqm) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:326]:
```suggestion
Note that these setups are mutually exclusive and cannot be configured together.
```

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 1 above)
Comment (HarshalNeelkamal) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Done

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 2 above)
Comment (HarshalNeelkamal) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Done

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 3 above)
Comment (HarshalNeelkamal) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Done

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 4 above)
Comment (HarshalNeelkamal) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Done

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 5 above)
Comment (HarshalNeelkamal) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Done

File: content/en/docs/reference/command-line-tools-reference/feature-gates/external-service-account-token-signer.md (hunk 6 above)
Comment (HarshalNeelkamal) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/external-service-account-token-signer.md]: Updated

File: content/en/docs/reference/command-line-tools-reference/feature-gates/external-service-account-token-signer.md (hunk 7 above)
Comment (HarshalNeelkamal) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/external-service-account-token-signer.md]: Done

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 8 above)
Comment (HarshalNeelkamal) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Done

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 9 above)
Comment (HarshalNeelkamal) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Done

**Comment by jihoon-seo** [comment]
/remove-area blog localization release-eng web-development
/remove-language bn de es fr hi id it ja ko pl pt ru uk vi zh
/remove-sig release

**Comment by liggitt** [comment]
tech content lgtm

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 10]
```diff
@@ -544,6 +550,20 @@ Then, delete the Secret you now know the name of:
...
+To configure kube-apiserver to use external-jwt-signer set the `--service-account-signing-endpoint` flag to the location of a Unix domain socket (UDS) on a filesystem, or be prefixed with an @ symbol and name a UDS in the abstract socket namespace.
+At the configured UDS, shall be an RPC server which implements [ExternalJWTSigner](https://github.com/kubernetes/kubernetes/blob/release-1.32/staging/src/k8s.io/externaljwt/apis/v1alpha1/api.proto).
+The external-jwt-signer must be healthy and be ready to serve supported service account keys for the kube-apiserver to start.
+
+Check out [KEP-740](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/740-service-account-external-signing) for more details on ExternalJWTSigner.
+
+NOTE: The kube-apiserver flags --service-account-key-file and --service-account-signing-key-file will continue to be used for reading from files unless --service-account-signing-endpoint is set; they are mutually exclusive ways of supporting JWT signing and authentication.
```
Comment (tengqm) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:565]: use the `note` shortcode?
consider wrap the flags with backticks?

**Comment by liggitt** [comment]
/lgtm

needs docs approval

**Comment by tengqm** [comment]
There are some doc style issues to fix but none of them are blockers for this to merge.
/approve

**Pull request merged**